feat(svgeople): Create svgeople

jefflembeck opened the pull request:
The svgeople component allows you add and remove inline svg images of
people to a panel. This will be useful for organization and enterprise
order counters

jefflembeck committed:
feat(svgeople): Create svgeople

The svgeople component allows you add and remove inline svg images of
people to a panel. This will be useful for organization and enterprise
order counters

aredridel commented on src/pivotal-ui/components/svgeople/index.js:
```diff
@@ -0,0 +1,103 @@
+};
+
+SVGeople.prototype.addPerson = function(idx) {
+  var id = "#svgerson_" + idx;
```
What happens when you instantiate this component multiple times? duplicate IDs?

aredridel replied:
What about getting the nth that matches a selector and removing that? Working by actual index rather than a stored ID?

aredridel commented on src/pivotal-ui/components/svgeople/svgeople.scss:
```diff
@@ -0,0 +1,117 @@
+      order: 1;
+    }
+
+    &:nth-child(10) {
```
Does it go to eleven?

jefflembeck replied:
Maxes out at 10 right now, since that is the current maximum for this thing, I'm going to leave it at that and wait for a later version to fix expanding it in a nicer way. Otherwise, everything will tack on at the end, which isn't the end of the world.

aredridel replied:
SGTM. document?

aredridel commented on src/pivotal-ui/components/svgeople/index.js:
```diff
@@ -0,0 +1,103 @@
+};
+
+SVGeople.prototype.fillPeople = function() {
+  var args = Array.prototype.slice.call(arguments);
```
Why not just take an array?

aredridel commented on src/pivotal-ui/components/svgeople/index.js:
```diff
@@ -0,0 +1,103 @@
+  }).join('');
+
+  this.el.html(people);
+  this.el.trigger("SVGeople:peopleAddedToContainer");
```
Hm. What's the use of this kind of event?

aredridel commented on src/pivotal-ui/components/svgeople/index.js:
```diff
@@ -0,0 +1,103 @@
+  var svgeople = new SVGeople(el);
+  svgeople.populate();
+
+  $(document).on("FormStepper:changed", function(e, number) {
```
This is awful purpose-specific is it not?

jefflembeck committed:
fix(svgeople): Documentation and API

These are a load of documentation and API changes, since these
changes are breaking in nature (removed public functions/triggered events),
this is a major version bump.

BREAKING CHANGE: If you were relying upon `fillPeople`, it no longer
exists, nor does the `this.people` array

jefflembeck committed:
chore(pui): update svgeople dep

jefflembeck committed:
fix(svgeople): code clean + docs

Further document and clean the way the code for svgeople works. Remove
duplicate ids and also make sure the reason for only 10 children being
handled is in place

jefflembeck committed:
chore(pui): update svgeople

aredridel commented:
Oh now that is shiny!

aredridel commented:
🐑
